modin/data_management/partitioning/partition_collections.py

```diff
@@ -40,6 +40,30 @@ def __init__(self, partitions):
     # Partition class is the class to use for storing each partition. It must
     # extend the `RemotePartition` class.
     _partition_class = None
+    # Whether or not we have already filtered out the empty partitions.
+    _filtered_empties = False
+
+    def _get_partitions(self):
+        if not self._filtered_empties:
+            self._partitions_cache = np.array(
+                [
+                    [
+                        self._partitions_cache[i][j]
+                        for j in range(len(self._partitions_cache[i]))
+                        if self.block_lengths[i] != 0 or self.block_widths[j] != 0
```

Comment:
Should this be an `and` instead of `or`? Or can a valid partition have one of the dimensions be 0?

Reply:
Yes valid partitions can have a dimension that is 0. If we filter both, we can end up losing indexing information we need for operations like `query`. It's annoying, but it has to be this way.